nodeadm/internal/api/status.go

```diff
@@ -30,9 +31,12 @@ func GetInstanceDetails(ctx context.Context, imdsClient *imds.Client, ec2Client
 		return nil, err
 	}
 
-	privateDNSName, err := getPrivateDNSName(ec2Client, instanceIdenitityDocument.InstanceID)
-	if err != nil {
-		return nil, err
+	privateDNSName := ""
+	if featuregates.DefaultTrue(featuregates.InstanceIdNamedNode, featureGates) {
```

Comment:
wait a second, this is an opt-in behavior so we should be using `DefaultFalse` in both usages

```suggestion
	if !featuregates.DefaultFalse(featuregates.InstanceIdNamedNode, featureGates) {
```

Reply:
kk, `! + defaultFalse` should be more readable

Reply:
its also behavioral since `DefaultTrue == !set || enabled` != `!DefaultFalse == !(set && enabled) == !set || !enabled` 😢